### PR TITLE
Dockerize API executable and localstack use for testing 

### PR DESCRIPTION
**Pull request by aaron-collier**
This is based on the taco docker work. This will also move us towards a standard development/testing stack and working with circle eventually.

This may also allow easier sharing of our work back to the community.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      context: .
+      dockerfile: Dockerfile.tester
+    environment:
+      AWS_ACCESS_KEY_ID: 999999
+      AWS_SECRET_ACCESS_KEY: 999999
+      AWS_DEFAULT_REGION: us-east-1
+      AWS_REGION: us-east-1
+      LOCAL_ENDPOINT_HOST: localstack
+    working_dir: /go/src/github.com/sul-dlss-labs/rialto-api
+    command: make resources
```
Comment (jcoyne) [inline docker-compose.yml]: this references a Makefile that doesn't exist.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on:
+      - localstack
+  tester:
+    build:
+      context: .
+      dockerfile: Dockerfile.tester
+    environment:
+      TEST_REMOTE_ENDPOINT: rialto-api:8080
+    working_dir: /go/src/github.com/sul-dlss-labs/rialto-api
+    command: make compose_tests
```
Comment (jcoyne) [inline docker-compose.yml]: Here too.

File: docker-compose.yml (hunk 1 above)
Comment (aaron-collier) [reply docker-compose.yml]: Yup, this was mostly copy past so far, cleaning up, but wanted to get this WIP on the radar :-) thanks for catching this.

File: docker-compose.yml [hunk 3]
```diff
@@ -21,6 +21,7 @@ services:
       - 4567-4583:4567-4583
     environment:
       SERVICES: lambda,es,sns
+      LAMBDA_EXECUTOR: docker
```
Comment (jcoyne) [inline docker-compose.yml]: Why are we even starting lambda?

Comment (aaron-collier) [reply docker-compose.yml]: I'm happy to move this off to rialto-lambda if that makes sense. The division wasn't there when I started this. But, I'm a little concerned about having to start local stack in multiple starts for consistent development environments... That said, I'm open to whatever we all think.

**Pull request merged**